### DISCUSSION TITLE
Add onFirstCodeKeyDown function

### DISCUSSION THREAD

**Discussion by smartland007** [HichemTab-tech/OTP-designer-jquery#7] [Ideas]
Is it possible to add a new function like:
onFirstCodeKeyDown | Function | A callback function executed when the user types the first code.

This will allow for example to call a function to hide an error message on the screen. The use case is the following:
- the user types a wrong code, we show a div with an error message
- we clear the OTP and call the focus function (ready to type the new code)
- when the user start to type the new code in the first cell, we would like to hide the error message

**Comment by HichemTab-tech** [comment]
Hello @smartland007,

Firstly, thank you for your this suggestions. Your feedback helps us make the library even more versatile and user-friendly.

Regarding your suggestion to add the `onFirstCodeKeyDown` function, we appreciate your creative use case. We've taken your suggestion into account and have made further improvements to the library.

In our latest update, we've introduced the `onchange` callback function, which is executed whenever the OTP code changes—whether through typing, clearing, or setting. This new functionality provides a way to achieve the behavior you described. To address your specific use case of hiding an error message when the user starts typing a new code, you can utilize the `onchange` function in the following manner:

```javascript
$().otpdesigner('option', 'onchange', function () {
    // Code to remove the error message here
    
    // Once the action is done, set 'onchange' back to null
    $().otpdesigner('option', 'onchange', null);
});
```

Once again, we greatly appreciate your feedback and hope these updates align with your needs. Please feel free to reach out if you have any further questions or suggestions.

Best regards,
@HichemTab-tech

**Comment by smartland007** [comment]
Hello,

Excellent !! Thank you very much.

Would it be better with a camel-case typo like: onChange instead
of onchange ?

Regards,
